src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -200,14 +200,16 @@ public void Enqueue(TElement element, TPriority priority)
             // Note that the node being enqueued does not need to be physically placed
             // there at this point, as such an assignment would be redundant.
 
-            int currentSize = _size++;
+            int currentSize = _size;
```

Comment:
could MoveUpDefaultComparer/MoveUpCustomComparer throw? If so should you be doing `_size + 1` here and `_size = currentSize` in the end of this method?

Reply:
No, it cannot. I could have moved it to the end of the method, but the Move* implementations currently require the size being up to date and I didn't want to change that invariant.